### PR TITLE
LPS-162050 User facet bucket ordering

### PR DESCRIPTION
**Pull request by AlmirFe**
https://issues.liferay.com/browse/LPS-162050

**Important:** This PR will be used for both frontend and backend tasks. If you are only reviewing the frontend, do not run ci:forward

### CODE REVIEW

**Comment by AlmirFe** [comment]
ci:test:relevant

**Comment by AlmirFe** [comment]
ci:test:sf

**Comment by AlmirFe** [comment]
ci:test:search

**Comment by AlmirFe** [comment]
ci:test:relevant

File: modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/facets/view/users.jsp [hunk 1]
```diff
@@ -22,6 +22,7 @@ UserSearchFacetDisplayContextBuilder userSearchFacetDisplayContextBuilder = new
 userSearchFacetDisplayContextBuilder.setFacet(facet);
 userSearchFacetDisplayContextBuilder.setFrequenciesVisible(dataJSONObject.getBoolean("showAssetCount", true));
 userSearchFacetDisplayContextBuilder.setFrequencyThreshold(dataJSONObject.getInt("frequencyThreshold"));
+userSearchFacetDisplayContextBuilder.setOrder(dataJSONObject.getString("order"));
 userSearchFacetDisplayContextBuilder.setMaxTerms(dataJSONObject.getInt("maxTerms", 10));
```
Comment (thektan) [inline modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/facets/view/users.jsp:26]: Can you alphabetically sort these lines?

```suggestion
userSearchFacetDisplayContextBuilder.setMaxTerms(dataJSONObject.getInt("maxTerms", 10));
userSearchFacetDisplayContextBuilder.setOrder(dataJSONObject.getString("order"));
```

File: modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/user/facet/configuration.jsp [hunk 2]
```diff
@@ -85,6 +85,13 @@ UserFacetPortletPreferences userFacetPortletPreferences = new UserFacetPortletPr
 				<aui:input label="frequency-threshold" name="<%= PortletPreferencesJspUtil.getInputName(UserFacetPortletPreferences.PREFERENCE_KEY_FREQUENCY_THRESHOLD) %>" value="<%= userFacetPortletPreferences.getFrequencyThreshold() %>" />
 
 				<aui:input label="display-frequencies" name="<%= PortletPreferencesJspUtil.getInputName(UserFacetPortletPreferences.PREFERENCE_KEY_FREQUENCIES_VISIBLE) %>" type="checkbox" value="<%= userFacetPortletPreferences.isFrequenciesVisible() %>" />
+
+				<aui:select label="order-terms-by" name="<%= PortletPreferencesJspUtil.getInputName(UserFacetPortletPreferences.PREFERENCE_KEY_ORDER) %>" value="<%= userFacetPortletPreferences.getOrder() %>">
+					<aui:option label="term-frequency-descending" value="count:desc" />
+					<aui:option label="term-frequency-ascending" value="count:asc" />
+					<aui:option label="term-value-ascending" value="key:asc" />
+					<aui:option label="term-value-descending" value="key:desc" />
+				</aui:select>
```
Comment (thektan) [inline modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/user/facet/configuration.jsp:94]: Can you move the "Order Terms By" select immediately after the "Frequency Threshold" field?

According to the acceptance criteria:

> Displayed as a simple dropdown called "Order Terms By" after the "Frequency Threshold" field

**Review by thektan — CHANGES_REQUESTED** [review]
Hi @AlmirFe! Thanks for these changes!

I requested a few places to re-order some of the code.

**Comment by AlmirFe** [comment]
> Hi @AlmirFe! Thanks for these changes!
> 
> I requested a few places to re-order some of the code.

I will work to make these changes, thanks @thektan

**Review by thektan — APPROVED** [review]
Thanks for making those changes! Frontend looks good 👍

**Comment by BryanEngler** [comment]
thanks for the pr, ive added the code to this branch:
https://github.com/BryanEngler/liferay-portal/commits/2022_10_04_LPS_161486_facet_ordering

once we are finished with https://issues.liferay.com/browse/LPS-164622 we will apply any necessary changes to this branch

**Pull request closed without merging**